### PR TITLE
feat(pip-compile): Provide credentials for registries in all input files

### PR DESCRIPTION
**Pull request by mbudnek**
## Changes

Pip-compile can compile multiple input files into one lock file. Renovate calls the manager's `updateArtifacts` function once for each input file, but currently it only looks for private package registries in the input file currently being processed.  This changes it to look for them in all of the input files.

## Context

#28958 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by mbudnek** [comment]
@viceice @rarkins Bump, just to make sure this PR hasn't fallen through the cracks.

File: lib/modules/manager/pip-compile/artifacts.ts [hunk 1]
```diff
@@ -108,12 +114,22 @@ export async function updateArtifacts({
       const compileArgs = extractHeaderCommand(existingOutput, outputFileName);
       const cwd = inferCommandExecDir(outputFileName, compileArgs.outputFile);
       const upgradePackages = updatedDeps.filter((dep) => dep.isLockfileUpdate);
-      const packageFile = pipRequirements.extractPackageFile(newInputContent);
+      const packageFiles: PackageFileContent[] = [];
+      for (const name of compileArgs.sourceFiles) {
+        const path = upath.join(cwd, name);
+        const content = await readLocalFile(path, 'utf8');
+        if (content) {
+          const packageFile = extractPackageFile(content, path, {});
```
Comment (viceice) [inline lib/modules/manager/pip-compile/artifacts.ts:122]: seems to be expensive, can't we split the extract function to smaller part's, so we only extract the relevant source urls instead of all dependencies again?

File: lib/modules/manager/pip-compile/extract.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ import {
   sortPackageFiles,
 } from './utils';
 
-function matchManager(filename: string): SupportedManagers | 'unknown' {
+export function matchManager(filename: string): SupportedManagers | 'unknown' {
```
Comment (viceice) [inline lib/modules/manager/pip-compile/extract.ts:26]: should be moved to common in separate PR

File: lib/modules/manager/pip_requirements/extract.ts [hunk 3]
```diff
@@ -47,11 +47,9 @@ export function cleanRegistryUrls(registryUrls: string[]): string[] {
   });
 }
 
-export function extractPackageFile(
+export function extractPackageFileFlags(
```
Comment (viceice) [inline lib/modules/manager/pip_requirements/extract.ts:50]: i would prefer to move this to a `common.ts` and directly import, so it's not a member of the public manager api.

Maybe we should do this as a refactor before this PR.

File: lib/modules/manager/pip-compile/common.ts [hunk 4]
```diff
@@ -267,6 +271,5 @@ export function getRegistryCredVarsFromPackageFile(
       ...hostCreds,
     };
   }
-
```
Comment (viceice) [inline lib/modules/manager/pip-compile/common.ts:270]: please avoid those changes

File: lib/modules/manager/pip-compile/extract.ts (hunk 2 above)
Comment (mbudnek) [reply lib/modules/manager/pip-compile/extract.ts:26]: Opened #29359

File: lib/modules/manager/pip_requirements/extract.ts (hunk 3 above)
Comment (mbudnek) [reply lib/modules/manager/pip_requirements/extract.ts:50]: Opened #29360

**Review by viceice — APPROVED** [review]

**Comment by mbudnek** [comment]
@viceice Did you mean to merge this when you approved it, or is there something blocking it?

**Review by secustor — APPROVED** [review]
